lib/src/model/documentation_comment.dart

```diff
@@ -14,22 +14,19 @@ import 'package:meta/meta.dart';
 import 'package:path/path.dart' as p show Context;
 
 final _templatePattern = RegExp(
-    r'[ ]*{@template\s+(.+?)}([\s\S]+?){@endtemplate}[ ]*(\n?)',
-    multiLine: true);
+    r'[ ]*\{@template\s+(\S.*?)\}([^]+?)\{@endtemplate\}[ ]*(\n?)');
```

Comment:
Added `\` before `{` and `}` just be cause it makes it clear they're not semantic.
The `[^]` is the same as `[\s\S]` (match *any* character).
The `multiLine: true` isn't used, there is no `^` or `$` in the regexp.

The `\s+(.+?)\}` pattern risks "linear backtracking" (not as bad as catastrophic backtracking, but still unnecessary).
An input of 
```

-- next line has many spaces -- 
{@template                                                                                      a
}
```  
will let all the space characters match `\s+`, then followed by an `a`, which must be matched by the `.`, and a newline which matches neither `.` nor `}`.
So it goes back and matches one less space in the `\s+`, lets `.` match that space instead, and continue to still not match. In the end, it tries all the combinations before giving up. 

Changing to `\s+([^\s}].*?)` makes the transition-point determinisitic, and also prevents matching `{@template }}` as having name `}`